### PR TITLE
[CIS-1194] Quoted Video Support

### PR DESCRIPTION
**Pull request by adamrushy**
### 🔗 Issue Link
https://stream-io.atlassian.net/browse/CIS-1194

### 🎯 Goal

Currently, if you inline reply with a Video you don't actually see anything quoted.

### 🛠 Implementation

We're taking a snapshot of the video by using Apple's AVKit API. Then set this along with the message contents.

### 🧪 Testing

1. Load the Demo App
2. Upload a Video
3. Reply to the Video Inline

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [x] Affected documentation updated (docusaurus, tutorial, CMS (task created)

### CODE REVIEW

**Comment by adamrushy** [comment]
It appears that QuotedMessage has no unit test coverage. I am going to create a technical ticket to cover all of this class with the required unit tests.

**Review by bielikb — APPROVED** [review]

File: Sources/StreamChatUI/CommonViews/QuotedChatMessageView/QuotedChatMessageView.swift [hunk 1]
```diff
@@ -238,6 +244,26 @@ open class QuotedChatMessageView: _View, ThemeProvider, SwiftUIRepresentable {
...
+            let generator = AVAssetImageGenerator(asset: asset)
+            generator.appliesPreferredTrackTransform = true
+            
+            // Create a CGImage from the generator
+            if let cgImage = try? generator.copyCGImage(at: CMTime(seconds: 2, preferredTimescale: 60), actualTime: nil) {
+                DispatchQueue.main.async { [weak self] in
+                    self?.attachmentPreviewView.image = UIImage(cgImage: cgImage)
+                }
+            }
+        }
```
Comment (nuno-vieira) [inline Sources/StreamChatUI/CommonViews/QuotedChatMessageView/QuotedChatMessageView.swift:265]: We already have a function in `VideoLoading` component that does exactly this, we should use the `components.videoLoader` to generate the Video Image here too 👍

**Review by nuno-vieira — CHANGES_REQUESTED** [review]
Requesting changes, since we already have a reusable component to generate video images 🙂

File: Sources/StreamChatUI/CommonViews/QuotedChatMessageView/QuotedChatMessageView.swift (hunk 1 above)
Comment (adamrushy) [reply Sources/StreamChatUI/CommonViews/QuotedChatMessageView/QuotedChatMessageView.swift:265]: Great spot @nuno-vieira pushing changes now 👍🏼

**Review by nuno-vieira — APPROVED** [review]
LGTM! ✅

File: Sources/StreamChatUI/CommonViews/QuotedChatMessageView/QuotedChatMessageView.swift [hunk 2]
```diff
@@ -238,6 +244,21 @@ open class QuotedChatMessageView: _View, ThemeProvider, SwiftUIRepresentable {
...
+    /// - Parameter url: The URL from which to generate the image on the video
+    open func setVideoAttachmentPreviewImage(url: URL?) {
+        guard let url = url else { return }
+        
+        components.videoLoader.loadPreviewForVideo(at: url) { [weak self] in
+            switch $0 {
+            case let .success(preview):
+                self?.attachmentPreviewView.image = preview
+            case .failure:
+                self?.attachmentPreviewView.image = nil
```
Comment (bielikb) [inline Sources/StreamChatUI/CommonViews/QuotedChatMessageView/QuotedChatMessageView.swift:258]: @adamrushy please log the error here.

Comment (adamrushy) [reply Sources/StreamChatUI/CommonViews/QuotedChatMessageView/QuotedChatMessageView.swift:258]: 👍🏼

**Pull request merged**